packages/frontend/src/components/ExploreTree.tsx

```diff
@@ -99,7 +99,7 @@ const TableTree: FC<TableTreeProps> = ({search, table, joinSql, selectedNodes, o
         id: table.name,
         label: friendlyName(table.name),
         isExpanded: expanded,
-        secondaryLabel: <TableButtons joinSql={joinSql} table={table} onOpenSourceDialog={onOpenSourceDialog}/>,
+        secondaryLabel: table.source && <TableButtons joinSql={joinSql} table={table} onOpenSourceDialog={onOpenSourceDialog}/>,
```

Comment:
We still want access to the `join details` menu. So this logic needs to be inside <TableButtons/>

Reply:
Good catch!!! 😅 

Didn't realise that the join had moved under the overflow menu